examples/query/react/basic/src/store.ts

```diff
@@ -1,11 +1,13 @@
 import { configureStore } from '@reduxjs/toolkit'
 import { pokemonApi } from './services/pokemon'
 
-export const store = configureStore({
-  reducer: {
-    [pokemonApi.reducerPath]: pokemonApi.reducer,
-  },
-  // adding the api middleware enables caching, invalidation, polling and other features of `rtk-query`
-  middleware: (getDefaultMiddleware) =>
-    getDefaultMiddleware().concat(pokemonApi.middleware),
-})
+export const setUpStore = () => {
+  return configureStore({
+    reducer: {
+      [pokemonApi.reducerPath]: pokemonApi.reducer,
+    },
+    // adding the api middleware enables caching, invalidation, polling and other features of `rtk-query`
+    middleware: (getDefaultMiddleware) =>
+      getDefaultMiddleware().concat(pokemonApi.middleware),
+  })
+}

```

Comment:
As the store should maybe also be used with non-RTKQ tests, we could already do a bit more setup here.

The question here is how far we go with that. 
Allowing for `preloadedState` would be great, but it requires that we call `combineReducers` ahead of time to get the correct type for that.

```suggestion
import {
  configureStore,
  combineReducers,
  PreloadedState,
} from '@reduxjs/toolkit'
import { pokemonApi } from './services/pokemon'

// Allow for a preloadedState.
// Doing so moves quite a few things around (we need the reducer scoped outside to get the RootState type which is required for the PreloadedState type), so I'm not 100% sure it's worth it, but many people would probably encounter the need for this.

const reducer = combineReducers({
  [pokemonApi.reducerPath]: pokemonApi.reducer,
})
export const setUpStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer,
    // adding the api middleware enables caching, invalidation, polling and other features of `rtk-query`
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(pokemonApi.middleware),
    preloadedState,
  })
}

// we should probably show how to get those types with this more testing-targeted setup
export type RootState = ReturnType<typeof reducer>
export type AppDispatch = ReturnType<typeof setUpStore>['dispatch']

```

Reply:
Passing `preloadedState` also makes bolting on Storybook super easy :)

Reply:
Sounds good! That's another pattern that I utilize at work, so I'll get that worked in. I definitely agree with being consistent and doing the best practices all around for all the test projects.